pygradle-plugin/src/main/groovy/com/linkedin/gradle/python/tasks/Flake8Task.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.gradle.python.tasks;
+
+import com.linkedin.gradle.python.PythonExtension;
+import com.linkedin.gradle.python.extension.PythonDetails;
+import com.linkedin.gradle.python.tasks.action.CreateVirtualEnvAction;
+import com.linkedin.gradle.python.tasks.action.PipInstallAction;
+import com.linkedin.gradle.python.tasks.exec.ProjectExternalExec;
+import com.linkedin.gradle.python.tasks.supports.SupportsPackageInfoSettings;
+import com.linkedin.gradle.python.util.DefaultEnvironmentMerger;
+import com.linkedin.gradle.python.util.PackageInfo;
+import com.linkedin.gradle.python.util.PackageSettings;
+import com.linkedin.gradle.python.wheel.EmptyWheelCache;
+import org.apache.commons.io.FileUtils;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.process.ExecResult;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class Flake8Task extends AbstractPythonInfrastructureDefaultTask implements SupportsPackageInfoSettings {
+
+    private static final Logger log = Logging.getLogger(Flake8Task.class);
+    private PackageSettings<PackageInfo> packageSettings;
+
+    @Override
+    public PythonDetails getPythonDetails() {
+        return new PythonDetails(getProject(), new File(getProject().getBuildDir(), "flake8-venv"));
+
+    }
+
+    public void preExecution() {
+        PythonDetails flake8Python = getPythonDetails();
+
+        File flake8Exec = flake8Python.getVirtualEnvironment().findExecutable("flake8");
+        if (!flake8Exec.exists() || !flake8Exec.isFile()) {
+
+            // We don't know what happened to this venv, so lets kill it
+            if (flake8Python.getVirtualEnv().exists()) {
+                FileUtils.deleteQuietly(flake8Python.getVirtualEnv());
+            }
+
+            CreateVirtualEnvAction action = new CreateVirtualEnvAction(getProject(), flake8Python);
+            action.buildVenv(null);
```

Comment:
The `null` needs to be replaced with the same configurable object that the standard venv uses.
Users who want to override _distutils.cfg_ to avoid hitting external network (as we do) from their CI pipeline need to do that for both virtualenvs: flake8 and standard.